src/communication/websocket/session.cpp

```diff
@@ -86,14 +100,76 @@ void Session::DoRead() {
       }));
 }
 
-void Session::OnRead(boost::beast::error_code ec, size_t /*bytes_transferred*/) {
+void Session::DoClose() {
+  ws_.async_close(boost::beast::websocket::close_code::normal,
+                  boost::asio::bind_executor(strand_, [shared_this = shared_from_this()](boost::beast::error_code ec) {
+                    shared_this->OnClose(ec);
+                  }));
+}
+
+void Session::OnClose(boost::beast::error_code ec) {
+  if (ec) {
+    return LogError(ec, "close");
+  }
+  connected_.store(false, std::memory_order_relaxed);
+}
+
+utils::BasicResult<std::string> Session::Authorize(const nlohmann::json &creds) {
+  if (!auth_.Authenticate(creds.at("username").get<std::string>(), creds.at("password").get<std::string>())) {
+    return {"Authentication failed!"};
+  }
+#ifndef MG_ENTERPRISE
```

Comment:
```suggestion
#ifdef MG_ENTERPRISE
```
Is what you probably meant.

Reply:
Can you remind me why do we have `MG_ENTERPRISE`, when we have only one binary?

Reply:
If people insist on having pure open-source package, they can build it with disabled `MG_ENTERPRISE`.
Authentication is part of the binary regardless, so we don't actually need to protect this with the flag.